core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -824,12 +836,18 @@ class KafkaApis(val requestChannel: RequestChannel,
     } else {
       val nonExistentTopics = topics -- topicResponses.map(_.topic).toSet
       val responsesForNonExistentTopics = nonExistentTopics.map { topic =>
-        if (topic == Topic.GroupMetadataTopicName) {
+        if (topic == GroupMetadataTopicName) {
```

Comment:
nit: since this logic seems the same in both paths, maybe we could consolidate the logic. For example:
```scala
if (isInternalTopic(topic)) {
  val topicMetadata = createInternalTopic(topic)
  if (topicMetadata.error == Errors.COORDINATOR_NOT_AVAILABLE)
     new MetadataResponse.TopicMetadata(Errors.INVALID_REPLICATION_FACTOR, topic, isInternal(topic), java.util.Collections.emptyList())
  else
    topicMetadata
}